auth/password_hash_test.go

```diff
@@ -64,3 +65,64 @@ func TestSetBcryptCost(t *testing.T) {
 	assert.Equal(t, bcryptDefaultCost, bcryptCost)
 	assert.True(t, bcryptCostChanged)
 }
+
+func BenchmarkCompareHashAndPassword(b *testing.B) {
+	// Get the bcrypt hash of the given password.
+	bcryptHash := func(password []byte) []byte {
+		hash, err := bcrypt.GenerateFromPassword(password, bcryptDefaultCost)
+		require.NoError(b, err)
+		return hash
+	}
+	// hashAndPassword represents the password hash pair.
+	type hashAndPassword struct {
+		hash     []byte
+		password []byte
+	}
+	// Setup cache of size N
+	const numPasswords = 100
+	cachedHashes = NewCache(numPasswords)
+
+	// Generate the test data and warm up the cache if required.
+	generateTestData := func(warmCache bool, key string) []hashAndPassword {
+		hashAndPasswords := make([]hashAndPassword, numPasswords)
+		for i := 0; i < numPasswords; i++ {
+			password := []byte(fmt.Sprintf("%s%d", key, i))
+			hashAndPassword := hashAndPassword{
+				hash:     bcryptHash(password),
+				password: password,
+			}
+			hashAndPasswords[i%numPasswords] = hashAndPassword
```

Comment:
The modulo is not doing anything here, i is always within range.

```suggestion
			hashAndPasswords[i] = hashAndPassword
```

Reply:
This change is included in the latest commit.